libs-scala/scala-utils/src/main/scala/com/daml/scalautil/nonempty/NonEmpty.scala

```diff
@@ -171,6 +210,11 @@ object NonEmptyColl extends NonEmptyCollInstances {
 
   implicit def traverse[F[_]](implicit F: Traverse[F]): Traverse[NonEmptyF[F, *]] =
     NonEmpty.substF(F)
+
+  // by requiring Monoid instead of Semigroup, we exclude intersection semigroups,
```

Comment:
Why require `Monoid` in the first place?

Reply:
Requiring `Monoid` excludes a category of `Semigroup` on collections that we definitely don't want to be lifted into `NonEmpty`, namely _intersection_ semigroups.

A typical monoid on collections has "union" like behavior, that is `append(a, b)` is always at least as large as the largest of a and b. These monoids are always desirable as semigroups for NonEmpty, because they preserve nonemptiness. However, we _don't_ want the full monoid to be lifted, because the zero of a union monoid is empty. By advertising semigroup instead, we eliminate the zero.

By contrast, consider the set-intersection semigroup. It is a semigroup--intersection is associative--but it can be applied to two non-empty sets and yield an empty set, which means we don't want it to be available at all for NonEmpty. Luckily, if you try to expand set-intersection to a monoid, you need to come up with something rather curious for the zero: a set with all values of the element type. So `mempty: Set[String]` is infinite, because it must intersect with all other `xs: Set[String]`s and still yield `xs`.

So if someone has defined an intersection semigroup for a collection type, it is unlikely that they have also defined it as a monoid.